code/game/gamemodes/cult/runes.dm

```diff
@@ -831,110 +680,54 @@ var/list/teleport_other_runes = list()
 		log_game("Talisman Imbue rune failed - no blank papers on rune")
 		return
 	var/obj/item/weapon/paper/paper_to_imbue = pick(papers_on_rune)
-	var/list/nearby_runes = list()
-	for(var/obj/effect/rune/R in orange(1,src))
-		nearby_runes.Add(R)
-	if(!nearby_runes.len)
-		user << "<span class='cultitalic'>There are no runes near [src]!</span>"
-		fail_invoke()
-		log_game("Talisman Imbue rune failed - no nearby runes")
-		return
-	var/obj/effect/rune/picked_rune = pick(nearby_runes)
-	var/list/split_rune_type = splittext("[picked_rune.type]", "/")
-	var/imbue_type = split_rune_type[split_rune_type.len]
-	var/talisman_type = text2path("/obj/item/weapon/paper/talisman/[imbue_type]")
-	if(ispath(talisman_type))
-		user.say("H'drak v'loso, mir'kanas verbot!")
-		var/obj/item/weapon/paper/talisman/TA = new talisman_type(get_turf(src))
-		if(istype(picked_rune, /obj/effect/rune/teleport))
-			var/obj/effect/rune/teleport/TR = picked_rune
-			var/obj/item/weapon/paper/talisman/teleport/TT = TA
-			TT.keyword = TR.keyword
-	else
-		user << "<span class='cultitalic'>The chosen rune was not a valid target!</span>"
-		fail_invoke()
-		log_game("Talisman Imbue rune failed - chosen rune invalid")
+	for(var/I in subtypesof(/obj/item/weapon/paper/talisman) - /obj/item/weapon/paper/talisman/malformed - /obj/item/weapon/paper/talisman/supply)
+		var/obj/item/weapon/paper/talisman/J = I
+		if(initial(J.cultist_name))
+			possible_talismans.Add(initial(J.cultist_name)) //This is to allow the menu to let cultists select talismans by name
```

Comment:
swap ```if(initial(J.cultist_name))```

for:

```
var/talisman_cult_name = initial(J.cultist_name)
if(talisman_cult_name)
```
then make this line  
```possible_talismans[talisman_cult_name] = J```

and then when you want to get the type just do ```talisman_type = possible_talismans[entered_talisman_name]```